enterprise/ha/src/main/java/org/neo4j/kernel/ha/lock/SlaveLocksClient.java

```diff
@@ -95,54 +97,75 @@ public SlaveLocksClient(
     }
 
     @Override
-    public void acquireShared( Locks.ResourceType resourceType, long resourceId ) throws AcquireLockTimeoutException
+    public void acquireShared( Locks.ResourceType resourceType, long... resourceIds ) throws AcquireLockTimeoutException
     {
         assertNotStopped();
 
         Map<Long, AtomicInteger> lockMap = getLockMap( sharedLocks, resourceType );
-        AtomicInteger preExistingLock = lockMap.get( resourceId );
-        if ( preExistingLock != null )
+        long[] newResourceIds = onlyFirstTimeLocks( lockMap, resourceIds );
+        if ( newResourceIds.length > 0 )
         {
-            // We already hold this lock, just increment the local reference count
-            preExistingLock.incrementAndGet();
+            acquireSharedOnMaster( resourceType, newResourceIds );
+            for ( long resourceId : newResourceIds )
+            {
+                if ( client.trySharedLock( resourceType, resourceId ) )
```

Comment:
We already teach lock clients to acquire multiple resources and it looks like it make sense to update 'try' calls to do that as well. WDYT?